lib/spack/docs/packaging_guide.rst

```diff
@@ -2775,6 +2775,248 @@ Suppose a user invokes ``spack install`` like this:
 Spack will fail with a constraint violation, because the version of
 MPICH requested is too low for the ``mpi`` requirement in ``foo``.
 
+.. _package-attributes:
+
+------------------
+Package attributes
+------------------
+
+Packages can provide custom implementations of four attributes used
+by dependents to interrogate various aspects of what it provides:
+
++-------------+---------------------------------------------+--------------------------+
+| Attribute   | Purpose                                     | Default                  |
++=============+=============================================+==========================+
+| ``root``    | The installation directory for the package  | ``spec.prefix``          |
+|             |                                             |                          |
++-------------+---------------------------------------------+--------------------------+
+| ``command`` | An executable command for the package       | ``spec.name`` in         |
+|             |                                             | ``spec.prefix.bin``      |
++-------------+---------------------------------------------+--------------------------+
+| ``headers`` | A list of headers provided by the package   | All headers in           |
+|             |                                             | ``spec.prefix.include``  |
++-------------+---------------------------------------------+--------------------------+
+| ``libs``    | A list of libraries provided by the package | ``lib{spec.name}`` in    |
+|             |                                             | ``spec.prefix``          |
```

Comment:
Not sure how to convey that this could be `spec.prefix.lib` or `spec.prefix.lib64`, or something else recursively

Reply:
I'll add some wording according.